src/interactive/azext_interactive/azclishell/recommendation.py

```diff
@@ -0,0 +1,392 @@
+from azure.cli.core import __version__ as version
+from azure.cli.core.style import print_styled_text, Style
+from prompt_toolkit.history import FileHistory
+
+
+class RecommendType(int, Enum):
+    All = 1
+    Solution = 2
+    Command = 3
+    Scenario = 4
+
+
+class RecommendThread(threading.Thread):
+    """ Worker Thread to fetch recommendation online based on user's context """
+    def __init__(self, cli_ctx, rec_path, executing_command, on_prepared_callback):
+        super().__init__()
+        self.cli_ctx = cli_ctx
+        self.on_prepared_callback = on_prepared_callback
+        # The latest 25 commands are extracted for personalized analysis
+        self.command_history = rec_path.get_cmd_history(25)
+        if executing_command:
+            self.command_history.append(executing_command)
+        self.processed_exception = rec_path.get_result_summary() if not executing_command else None
+        self.result = None
+
+    def run(self) -> None:
+        try:
+            self.result = get_recommend_from_api([json.dumps(cmd) for cmd in self.command_history], 1,
+                                                 self.cli_ctx.config.getint('next', 'num_limit', fallback=5),
+                                                 error_info=self.processed_exception)
+        except RecommendationError:
+            self.result = []
+        self.on_prepared_callback()
+
+
+class Recommender:
+    def __init__(self, cli_ctx, filename):
+        self.cli_ctx = cli_ctx
+        self.rec_path = RecommendPath(filename)
+        self.cur_thread = None
+        self.on_recommendation_prepared = lambda: None
+        self.default_recommendations = {
+            'help': 'Get help message of Azure CLI',
+            'init': 'Set Azure CLI global configurations interactively',
+            'next': 'Recommend the possible next set of commands to take'
+        }
+        self.executing_command = None
+
+    @property
+    def enabled(self):
+        """Whether recommender is enabled in global config"""
+        return self.cli_ctx.config.getboolean("interactive", "enable_recommender", fallback=True)
+
+    def _feedback_command(self, command):
+        """Send user's command choice in recommendations to telemetry."""
+        if self.cur_thread and not self.cur_thread.is_alive():
+            recommendations = self.cur_thread.result
+            trigger_commands = [cmd['command'] for cmd in self.cur_thread.command_history[-2:]]
+            processed_exception = self.cur_thread.processed_exception
+            if not recommendations or not command:
+                send_feedback(-1, trigger_commands, processed_exception, recommendations, rec=None)
+                return
+            for idx, rec in enumerate(recommendations):
+                if rec['type'] != RecommendType.Command:
+                    continue
+                if re.sub(r'^az ', '', re.sub(r'\s+', ' ', command)).strip().startswith(rec['command']):
```

Comment:
The judgment logic is complex. Please split it into variables with better readability and add examples and comments for the regular expression

Reply:
Done